test/model/message_test.dart

```diff
@@ -273,75 +273,122 @@ void main() {
       });
 
       test('message topic both resolved and edited update', () async {
-        await sendEvent(message, eg.updateMessageMoveEvent([message, otherMessage],
-          origTopic: 'new topic',
+        await prepareOrigMessages(origTopic: 'new topic');
+        await store.handleEvent(eg.updateMessageEventMoveFrom(
+          origMessages: origMessages,
           newTopic:  '✔ new topic 2'));
+        checkNotified(count: 2);
         check(store).messages.values.every(((message) => message.editState.equals(MessageEditState.moved)));
       });
 
       test('message topic both unresolved and edited update', () async {
-        await sendEvent(message, eg.updateMessageMoveEvent([message, otherMessage],
-          origTopic: '✔ new topic',
+        await prepareOrigMessages(origTopic: '✔ new topic');
+        await store.handleEvent(eg.updateMessageEventMoveFrom(
+          origMessages: origMessages,
           newTopic:  'new topic 2'));
+        checkNotified(count: 2);
         check(store).messages.values.every(((message) => message.editState.equals(MessageEditState.moved)));
       });
 
       test('message stream moved update', () async {
-        await sendEvent(message, eg.updateMessageMoveEvent([message, otherMessage],
-          origTopic: 'topic',
-          newTopic: 'topic',
-          newStreamId:  20));
-        check(store).messages.values.every(((message) => message.editState.equals(MessageEditState.moved)));
+        await prepareOrigMessages(origTopic: 'topic');
+        await store.handleEvent(eg.updateMessageEventMoveFrom(
+          origMessages: origMessages,
+          newTopic:  'topic-new',
+          newStreamId: 20));
+        checkNotified(count: 2);
+        check(store).messages.values.every(((message) =>
+          message.isA<StreamMessage>()
+            ..editState.equals(MessageEditState.moved)
+            ..displayRecipient.equals(null)));
+      });
+
+      test('message stream moved without topic change', () async {
+        await prepareOrigMessages(origTopic: 'topic');
+        await store.handleEvent(eg.updateMessageEventMoveFrom(
+          origMessages: origMessages,
+          newStreamId: 20));
+        checkNotified(count: 2);
+        check(store).messages.values.every(((message) =>
+          message.isA<StreamMessage>()
+            ..editState.equals(MessageEditState.moved)
+            ..displayRecipient.equals(null)));
       });
 
       test('message is both moved and updated', () async {
-        await sendEvent(message, eg.updateMessageMoveEvent([message, otherMessage],
-          origTopic: 'topic',
-          newTopic: 'topic',
-          newStreamId:  20,
-          origContent: 'old content',
+        await prepareOrigMessages(origTopic: 'topic', origContent: 'old content');
+        await store.handleEvent(eg.updateMessageEventMoveFrom(
+          origMessages: origMessages,
+          newTopic:  'new topic',
```

Comment:
This `newTopic` line changes in the commit:

    model: Expect the absence of newTopic for certain moves.

Are those changes connected? Does the new topic here need to change at all — could it just remain the way the test was, where the topic doesn't get renamed but the stream ID changes?

Reply:
Right, the tests changes shouldn't be necessary.